Source/WebCore/rendering/GridBaselineAlignment.cpp

```diff
@@ -56,16 +57,18 @@ LayoutUnit GridBaselineAlignment::marginUnderForChild(const RenderBox& child, Gr
     return isHorizontalBaselineAxis(axis) ? child.marginLeft() : child.marginBottom();
 }
 
-LayoutUnit GridBaselineAlignment::logicalAscentForChild(const RenderBox& child, GridAxis baselineAxis) const
+LayoutUnit GridBaselineAlignment::logicalAscentForChild(const RenderBox& child, GridAxis baselineAxis, ItemPosition position) const
 {
-    LayoutUnit ascent = ascentForChild(child, baselineAxis);
-    return isDescentBaselineForChild(child, baselineAxis) ? descentForChild(child, ascent, baselineAxis) : ascent;
+    LayoutUnit ascent = ascentForChild(child, baselineAxis, position);
+    return (isDescentBaselineForChild(child, baselineAxis) || position == ItemPosition::LastBaseline) ? descentForChild(child, ascent, baselineAxis) : ascent;
```

Comment:
I'm really struggling to get my head around the existing logic here, particularly the implementation of `isDescentBaselineForChild`. Why is it only considering the physical horizontal direction?

Are you sure that the logical OR of the two conditions is correct, not XOR?

I realise most of this isn't new to your changes, but if you understand it well enough to add comments explaining the physical/logical coord switches, then that'd be super helpful for me.

Reply:
isDescentBaselineForChild is a little hard for me to read as well. If I'm understanding it correctly, I don't think it's considering only the physical horizontal direction and it might be that isHorizontalBaselineAxis is misleadingly named?

If the baseline alignment axis is the column axis, then I think isHorizontalBaselineAxis will still return true if the grid is in a vertical writing mode, right? I'm still not sure why this requires us to use the descent value for alignment.